### PR TITLE
WIP: Refactor hybrid planning manager

### PR DESCRIPTION
**Pull request by sjahr**
### Description
I've refactored the Hybrid Planning Manager Class to avoid the ugly hacks @tylerjw [pointed out](https://github.com/ros-planning/moveit2/pull/813#pullrequestreview-818293852). The problem I encountered is that it is not straightforward to convert the Manager into a Node Class because it shares its ROS 2 communication interfaces with the planner logic plugin. I've tried to fix this by moving the interfaces into a second class called HybridPlannerInterface and thus avoid giving a shared_pointer of the HybridPlanningManager to the PlannerLogicPlugin. 

Currently, this does not work because a segfault error occurs at runtime which needs to be debugged :/

### Checklist
- [X] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Fix Segfault

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_ros/hybrid_planning/hybrid_planning_manager/hybrid_planning_manager_component/src/hybrid_planner_interface.cpp [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+          result->error_message = reaction_result.error_message;
+
+          hybrid_planning_goal_handle_->abort(result);
+          RCLCPP_ERROR(LOGGER, "Hybrid Planning Manager failed to react to  '%s'", reaction_result.event.c_str());
+        }
+      };
+
+  // Forward global trajectory goal from hybrid planning request TODO(sjahr) pass goal as function argument
+  auto global_goal_msg = moveit_msgs::action::GlobalPlanner::Goal();
+  global_goal_msg.motion_sequence =
```
Comment (sjahr) [inline moveit_ros/hybrid_planning/hybrid_planning_manager/hybrid_planning_manager_component/src/hybrid_planner_interface.cpp:156]: This line causes a segfault

File: moveit_ros/hybrid_planning/hybrid_planning_manager/planner_logic_plugins/include/moveit/planner_logic_plugins/single_plan_execution.h [hunk 2]
```diff
@@ -47,7 +46,7 @@ class SinglePlanExecution : public PlannerLogicInterface
 public:
   SinglePlanExecution() = default;
   ~SinglePlanExecution() = default;
-  bool initialize(const std::shared_ptr<HybridPlanningManager>& hybrid_planning_manager) override;
+  bool initialize(const std::shared_ptr<HybridPlannerInterface>& hybrid_planner_interface) override;
```
Comment (tylerjw) [inline moveit_ros/hybrid_planning/hybrid_planning_manager/planner_logic_plugins/include/moveit/planner_logic_plugins/single_plan_execution.h:49]: Locally I tried to solve the circular relationship here by giving a non-owning raw pointer to HybridPlanningManager through this interface.

The problem before is that these `PlannerLogicInterface` classes had a shared pointer to HybridPlanningManager which owned a list of `PlannerLogicInterface` objects.  What this means is that that they share ownership of each other causing problems when they are destroyed (causing that bad weak ptr exception).  The reason this was not caught before this change is we created the shared pointer to HybridPlannerInterface in a thread (timer callback) so it was created after HybridPlannerInterface finished initializing.

File: moveit_ros/hybrid_planning/hybrid_planning_manager/hybrid_planning_manager_component/src/hybrid_planning_manager.cpp [hunk 3]
```diff
@@ -81,19 +59,19 @@ bool HybridPlanningManager::initialize()
...
-    logic_plugin_name = this->declare_parameter<std::string>("planner_logic_plugin_name",
-                                                             "moveit::hybrid_planning/ReplanInvalidatedTrajectory");
+    logic_plugin_name = node_->declare_parameter<std::string>("planner_logic_plugin_name",
+                                                              "moveit::hybrid_planning/ReplanInvalidatedTrajectory");
   }
   try
   {
     planner_logic_instance_ = planner_logic_plugin_loader_->createUniqueInstance(logic_plugin_name);
-    if (!planner_logic_instance_->initialize(HybridPlanningManager::shared_from_this()))
+    if (!planner_logic_instance_->initialize(hybrid_planner_interface_))
```
Comment (tylerjw) [inline moveit_ros/hybrid_planning/hybrid_planning_manager/hybrid_planning_manager_component/src/hybrid_planning_manager.cpp:74]: Is `hybrid_planner_interface_` initialized at this point?  It looks to me that it isn't initialized until line 87.

File: moveit_ros/hybrid_planning/hybrid_planning_manager/hybrid_planning_manager_component/src/hybrid_planner_interface.cpp (hunk 1 above)
Comment (tylerjw) [reply moveit_ros/hybrid_planning/hybrid_planning_manager/hybrid_planning_manager_component/src/hybrid_planner_interface.cpp:156]: Where is the pointer `hybrid_planning_goal_handle_` initialized?

File: moveit_ros/hybrid_planning/hybrid_planning_manager/hybrid_planning_manager_component/src/hybrid_planning_manager.cpp (hunk 3 above)
Comment (sjahr) [reply moveit_ros/hybrid_planning/hybrid_planning_manager/hybrid_planning_manager_component/src/hybrid_planning_manager.cpp:74]: Yes :/ An uninitialized pointer is passed to the planner logic instance with the promise, that it won't be needed before it is initialized. This problem with the circular lifetime ownership I have not solved yet but I think it is better than passing the whole hybrid planning manager

**Pull request closed without merging**